src/sys/select.rs

```diff
@@ -53,6 +53,20 @@ impl FdSet {
             *bits = 0
         }
     }
+
+    /// Finds the largest file descriptor in the set.
+    ///
+    /// Returns `None` if the set is empty.
+    pub fn largest(&self) -> Option<RawFd> {
```

Comment:
I'd rather have `largest()` be private. And in general unit tests should be in the same file as the functions themselves, it's only the larger integration tests that should be in `/tests`. Let's more the tests into this file at the bottom in a `test` module.